cmake/Modules/FindMKL.cmake

```diff
@@ -88,25 +88,41 @@ ELSE ("${SIZE_OF_VOIDP}" EQUAL 8)
   SET(iccvers "ia32")
   SET(mkl64s)
 ENDIF ("${SIZE_OF_VOIDP}" EQUAL 8)
-IF(CMAKE_COMPILER_IS_GNUCC)
-  IF ("${MKL_THREADING}" STREQUAL "TBB")
-    SET(mklthreads "mkl_tbb_thread")
-    SET(mklrtls "tbb")
-  ELSE()
-    SET(mklthreads "mkl_gnu_thread" "mkl_intel_thread")
-    SET(mklrtls "gomp" "iomp5")
-  ENDIF()
-  SET(mklifaces  "intel" "gf")
-ELSE(CMAKE_COMPILER_IS_GNUCC)
+
+IF(WIN32)
   IF ("${MKL_THREADING}" STREQUAL "TBB")
```

Comment:
We got rid of all TBB builds recently, perhaps it's a good opportunity to clean up those as well

Reply:
I suggest to clean up code in further, this PR is only impact Windows build.
If we can fix the performance regression. Maybe we still have chance to cherry pick to `release/2.4`.